### DISCUSSION TITLE
v1.10 Darker models on some devices

### DISCUSSION THREAD

**Discussion by lucadalli** [google/model-viewer#3631] [Q&A]
### Description

Models on my Xiaomi Mi 9T appear darker on v1.10 than they did on v1.9 and previous versions. Might be irrelevant but the material on the model shown below has a base color factor of `#cccccc`. Even with a factor of `#ffffff` there is a difference but it is not as pronounced.
| v1.10 | v1.9 |
|------|-------|
|<img src="https://user-images.githubusercontent.com/39853447/153165789-7f4803b0-17c5-4d43-a8ab-06ff6b9abd8b.jpg" width="250"> |<img src="https://user-images.githubusercontent.com/39853447/153165774-d5592071-e4b5-4ad4-b26a-8ebb2fefb198.jpg" width="250">|

#### Live Demo
http://modelviewer-tester.glitch.me/
Attached below are screenshots demonstrating the astronaut on v1.10 and v1.9. The difference is not as evident as the models above but still present.
| v1.10 | v1.9 |
|------|-------|
|<img src="https://user-images.githubusercontent.com/39853447/153169477-197b0bab-35cb-4a5a-b77b-bcf578979712.jpg" width="250">|<img src="https://user-images.githubusercontent.com/39853447/153169465-55df430f-6de0-4bbc-9730-a9c06daabdc0.jpg" width="250">|

### Version

- model-viewer: v1.10.1

### Browser Affected

- [x] Chrome, version: 97.0.4692.98
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by samaneh-kazemi** [comment]
Hi @lucadalli, thanks for reaching out. This is intentional. As of #3082 pr, we switched from gamma encoding to sRGB and that's causing the change. You should be able to get similar results to before by playing with exposure.

**Comment by lucadalli** [comment]
@samaneh-kazemi The release notes state that the intended behavior is for the models to appear _lighter_ not _darker_ with v1.10.
In fact, on my Windows machine models appear ever so slightly lighter, as expected. On the Xiaomi Mi 9T, models appear considerably darker (kindly revisit the screenshots above and pay a closer look to the version labels).
It's intentional that models appear slightly lighter on some devices and considerably darker on others?

**Comment by lucadalli** [comment]
Here's a comparison demonstrating the inconsistency between the two devices.
Observe how version v1.9 (right) appears the same on both devices, and v1.10 is slightly lighter on the Windows machine but darker on the Xiaomi Mi 9T.

| Windows v1.10 | Windows v1.9 | 
| --- | --- |
| <img src="https://user-images.githubusercontent.com/39853447/153308897-2724f8fa-35fb-464d-90c6-85b24f86c018.jpg" width="250"> | <img src="https://user-images.githubusercontent.com/39853447/153308901-13825649-4981-493a-8510-ea9a3404d465.jpg" width="250"> | 

| Xiaomi Mi 9T v1.10 | Xiaomi Mi 9T v1.9 |
| --- | --- | 
| <img src="https://user-images.githubusercontent.com/39853447/153165789-7f4803b0-17c5-4d43-a8ab-06ff6b9abd8b.jpg" width="250"> | <img src="https://user-images.githubusercontent.com/39853447/153165774-d5592071-e4b5-4ad4-b26a-8ebb2fefb198.jpg" width="250"> |

**Comment by elalish** [comment]
Now that is quite interesting! @mrdoob do you have any idea why the Xiaomi would be darker now on r136? I assume this is from changing `gammaEncoding` to `sRGBEncoding` which I did as part of the upgrade. Do you think this is some kind of driver bug, or maybe they just don't support `sRGB`? Have you noticed this on any of your devices?

**Comment by mrdoob** [comment]
My devices are still in storage in San Francisco 😢

@lucadalli could you check if your device supports WebGL 2.0? https://webglreport.com/?v=2

**Comment by lucadalli** [comment]
@mrdoob
<img src="https://user-images.githubusercontent.com/39853447/153313764-da4e0068-fc8f-46e1-bc02-e2b1c87bb57c.jpg" width="250">

**Comment by elalish** [comment]
@Mugen87 Do you happen to have any devices that repro this? It seems like some kind of sRGB driver problem? I haven't managed to repro.

**Comment by Mugen87** [comment]
I have compared `v1.10` with `v1.9` on three devices so far (iMac, Windows Laptop and Pixel4a) and all render as expected.

@lucadalli It seems you have only tested with Chrome on your Xiaomi Mi 9T. Can you please make a test with a different browser like Firefox? If the behavior is the same, it indeed indicates a driver issue.

For clarity, sRGB encoded textures used to darken in `three.js` which was wrong. Since `r136` the related sRGB workflow is implemented differently by using `SRGB8_ALPHA8` which fixed this issue. So depending on the texture filtering settings, textures appear lighter which just means they do no wrongly darken anymore (the brightness is now constant).

**Comment by Mugen87** [comment]
For better testing here are two fiddles with plain `three.js`:

r132 (❌): https://jsfiddle.net/sj50nrw4/1/
r138 (✅): https://jsfiddle.net/jtmnd9x4/

With `r132` textures should get slightly darker from left to right. With `r138`, brightness keeps constant.

If Chrome does not render like described on a device, I suggest to file a bug at the Chromium bug tracker.
